File: app/src/main/java/com/live/utils/DeviceUtil.java
package com.live.utils;

import android.util.DisplayMetrics;

import com.live.LiveApp;


/**
 * Created by gaoruochen on 18-6-4.
 */

public class DeviceUtil {

    public static int getScreenWidth() {
        DisplayMetrics dm;
        dm = LiveApp.getInstance().getResources().getDisplayMetrics();
        return dm.widthPixels;
    }

    public static int getScreenHeight() {
        DisplayMetrics dm;
        dm = LiveApp.getInstance().getResources().getDisplayMetrics();
        return dm.heightPixels;
    }

    public static int dip2px(float dpValue) {
        final float scale = LiveApp.getInstance().getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }


}

File: app/src/main/java/com/live/utils/DeviceUtil.kt
package com.live.utils

import android.util.DisplayMetrics

import com.live.LiveApp


/**
 * Created by gaoruochen on 18-6-4.
 */

object DeviceUtil {

    val screenWidth: Int
        get() {
            val dm: DisplayMetrics = LiveApp.instance!!.resources.displayMetrics
            return dm.widthPixels
        }

    val screenHeight: Int
        get() {
            val dm: DisplayMetrics = LiveApp.instance!!.resources.displayMetrics
            return dm.heightPixels
        }

    val systemVersion: String
        get() = android.os.Build.VERSION.RELEASE

    val systemModel: String
        get() = android.os.Build.MODEL

    fun dip2px(dpValue: Float): Int {
        val scale = LiveApp.instance!!.resources.displayMetrics.density
        return (dpValue * scale + 0.5f).toInt()
    }


}
